cmd/tools/db-seeder/pkg/seeder/air_quality/air_quality.go

```diff
@@ -0,0 +1,65 @@
+package air_quality
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smart-core-os/sc-api/go/traits"
+	"github.com/smart-core-os/sc-golang/pkg/trait"
+)
+
+func Seed(ctx context.Context, db *pgxpool.Pool, name string, lookBack time.Duration) error {
+	current := time.Now()
+	now := current
+
+	source := fmt.Sprintf("%s[%s]", name, trait.AirQualitySensor)
+
+	for current.After(now.Add(-lookBack)) {
+
+		co2 := rand.Float32() * 2000
+		voc := rand.Float32() * 500
+		airPressure := rand.Float32()
+		comfort := traits.AirQuality_Comfort(rand.Intn(2) + 1)
+		infection := rand.Float32() * 100
+		score := rand.Float32() * 100
+		particulate1 := rand.Float32()
+		particulate25 := rand.Float32()
+		particulate10 := rand.Float32()
+		airChange := rand.Float32() * 5
+
+		payload, err := proto.Marshal(&traits.AirQuality{
+			CarbonDioxideLevel:       &co2,
+			VolatileOrganicCompounds: &voc,
+			AirPressure:              &airPressure,
+			Comfort:                  comfort,
+			InfectionRisk:            &infection,
+			Score:                    &score,
+			ParticulateMatter_1:      &particulate1,
+			ParticulateMatter_10:     &particulate10,
+			ParticulateMatter_25:     &particulate25,
+			AirChangePerHour:         &airChange,
+		})
+
+		if err != nil {
+			return err
+		}
+		cmd, err := db.Exec(ctx, `INSERT INTO history (source, create_time, payload) VALUES ($1, $2, $3)`, source, current.Format(time.RFC3339Nano), payload)
```

Comment:
I'm not a big fan of embedding these queries directly into the tool - it couple the tool to the database schema.
Have you considered extending the `history/pgxstore` package to add what you need?

(same applies to the `air_quality.Seed` and `meter.Seed` too)

Reply:
I tried but history/pgxstore does it's own `RecordTime` values for create time and I didn't have time to go into that functionality and edit it. Writing my own query was MVP.

Reply:
I think the way to put this into `pgxstore` would be:
- Change `pgxstore.New`, `pgxstore.SetupStoreFromPool` and `pgxstore.NewStoreFromPool` to return a `*pgxstore.Store` instead of a `history.Store` (I think this is a good thing to do anyway - it's idiomatic in Go to [accept interfaces as arguments, and return structs](https://medium.com/@cep21/what-accept-interfaces-return-structs-means-in-go-2fe879e25ee8))
- Add a method `Insert(ctx context.Context, at time.Time, payload []byte) (history.Record, error)` to `pgxstore.Store` that allows specifying the time for the new record.
  - `Append` could then be trivially reimplemented in terms of `Insert`.

That way, you can specify the time and the SQL is still encapsulated in that package.